nixos/modules/services/networking/haproxy.nix

```diff
@@ -60,15 +60,31 @@ with lib;
       description = "HAProxy";
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
+      reloadIfChanged = true;
```

Comment:
I'm not sure you actual want this... I'm under the impression if a CVE were reported and fixed then you updated your OS the service would continue to run with the existing vulnerable binary. A manual stop and start would be required to get the newly patched binary.

Reply:
The reload mechanism implemented here actually starts a completely new process, and kills the old one. The transition is implemented through a socket which forwards the sockets between the two processes. There is no shared memory between the two processes, only a few environment variables are passed from the old master process to the new binaries.

I have manually tested that the old binary is gone after reloading, and I can try to write a NixOS test for that if it is necessary (comparing PIDs before/after reloading).

Reply:
@pstch thanks for mentioning that. I'll have to review some scripts as I was under a different impression.